lib/path.js

```diff
@@ -1,7 +1,14 @@
 'use strict';
 
+const util = require('util');
 const isWindows = process.platform === 'win32';
 
+function assertPath(path) {
+  if (typeof path !== 'string')
+    throw new TypeError('Path must be a string. Received ' +
+                        util.inspect(path));
+}
+
```

Comment:
Maybe something like this makes it more clear which argument is the wrong type:

````js
function assertPath() {
  [].slice.call(arguments).forEach(function (path, i) {
    if (typeof path !== 'string')
        throw new TypeError(util.format('Argument %d must be a string.' +
                            ' Received %s', i + 1, util.inspect(path));
  });
}
````